src/meta/src/hummock/compaction/tier_compaction_picker.rs

```diff
@@ -130,7 +131,10 @@ impl TierCompactionPicker {
         let mut new_add_tables = vec![];
         // pick up files in L1 which are overlap with L0 to target level input.
         for table in &level.table_infos {
-            if !self.overlap_strategy.check_overlap(select_table, table) {
+            if !self.check_key_range_overlap(
```

Comment:
hope that we can restore correctness of compaction

Reply:
Are they different?